types/nodemailer/index.d.ts

```diff
@@ -42,34 +42,34 @@ export interface TestAccount {
     web: string;
 }
 
-export function createTransport(
-    transport?: SMTPTransport | SMTPTransport.Options | string,
-    defaults?: SMTPTransport.Options,
-): Transporter<SMTPTransport.SentMessageInfo>;
```

Comment:
This typedef is moved down by several lines. 

I suspect the typedef coverage is too large, that the TS compiler cannot narrow the type down to the more specific func def with `SMTPPool.Options` or `SendmailTransport.Options`.

For Maintainer: please let me know if you think it is inappropriate then we may discuss further on how to get around with it.

Reply:
Are you worried about the previous version, the new version, or both versions having difficulty narrowing? 

I took a quick look at `SMTPPool.Options`, `SendmailTransport.Options`, and `SMTPTransport.Options`, and your reasoning for moving this looks correct to me. Is `SMTPTransport.Options` the least specific of all the overloads (and thus should be last)?
I don't see any changes with the tests, so adding test cases for `createTransport` that show the different behavior of this overload would be good, because it sounds like the previous behavior was incorrect.

Reply:
After studying the [library implementation](https://github.com/nodemailer/nodemailer/blob/master/lib/nodemailer.js#L39-L52), I think the reordering I made is correct.

I have added more explicit type annotation to each `createTransport`. Running the updated test file with the previous typedef would at least give you 1 error.
```
<repo-dir>/types/nodemailer/nodemailer-tests.ts
  464:5  error  TypeScript@4.8, 4.9, 5.0, 5.1, 5.2, 5.3, 5.4, 5.5, 5.6, 5.7 compile error: 
Type 'import("<repo-dir>/types/nodemailer/index").Transporter<import("<repo-dir>/types/nodemailer/lib/smtp-transport/index").SentMessageInfo, import("<repo-dir>/types/nodemailer/lib/smtp-transport/index").Options>' is not assignable to type 'import("<repo-dir>/types/nodemailer/index").Transporter<import("<repo-dir>/types/nodemailer/lib/smtp-pool/index").SentMessageInfo, import("<repo-dir>/types/nodemailer/lib/smtp-pool/index").Options>'.
  Type 'SentMessageInfo' is missing the following properties from type 'SentMessageInfo': envelopeTime, messageTime, messageSize  @definitelytyped/expect
```

Hope the overloaded type reordering and the test change make sense to you. I will also add these information to the PR description.

Reply:
Looks good, thanks for the clarification and tests!